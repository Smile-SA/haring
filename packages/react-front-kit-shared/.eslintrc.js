module.exports = {
  extends: ['custom'],
  parserOptions: {
    project: ['./tsconfig.json'],
    tsconfigRootDir: __dirname,
  },
  root: true,
};
