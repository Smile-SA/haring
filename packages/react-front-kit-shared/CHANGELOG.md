# @smile/react-front-kit-shared

## 0.4.0

### Minor Changes

- [#53](https://github.com/Smile-SA/react-front-kit/pull/53) [`01ee3ea`](https://github.com/Smile-SA/react-front-kit/commit/01ee3ea74827cb6e3d4041b50fbab959750135ff) Thanks [@tonai](https://github.com/tonai)! - Use `Paper` for `InfoCard`, update test page and refactor `Pagination` `itemsPerPage` prop

## 0.3.0
