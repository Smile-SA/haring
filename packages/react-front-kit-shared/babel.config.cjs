const { babelConfig } = require('test');

module.exports = babelConfig;
