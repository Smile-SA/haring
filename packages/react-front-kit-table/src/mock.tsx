/* eslint-disable react-refresh/only-export-components */
export * from './Components/Table/Table.mock';
export * from './Components/TableGridView/TableGridView.mock';
