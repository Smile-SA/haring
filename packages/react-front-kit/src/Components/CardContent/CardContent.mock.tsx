import {
  ActionIcon,
  Avatar,
  Button,
  Flex,
  Indicator,
  Text,
} from '@mantine/core';
import { Export, Eye, UserCircle } from '@phosphor-icons/react';

export const defaultCardContentMock = {
  children: 'children node',
  leftNode: 'left node',
  rightNode: 'right node',
  titleNode: 'title node',
  topNode: 'top node',
};

export const topMock = (
  <span style={{ color: '#5C5F66', fontSize: 12 }}>30th novembre 2023</span>
);

export const otherTopMock = (
  <span style={{ color: '#5C5F66', fontSize: 12 }}>3 hours ago</span>
);

export const titleMock = <b>Lorem ipsum dolor sit amet</b>;

export const otherTitleMock = (
  <span>
    <b>Lorem ipsum</b>
    <span>
      {' '}
      dolor sit amet consectetur. Sollicitudin mattis blandit aliquet odio urna
      mi id :{' '}
    </span>
    <Text c="blue" component="span">
      Lorem Ipsum dolor
    </Text>
  </span>
);

export const childrenExampleMock = (
  <Flex direction="column" gap={24}>
    <span>
      Lorem ipsum dolor sit amet consectetur. Sollicitudin mattis blandit
      aliquet odio urna mi id. Mauris venenatis ut et at amet vel est.
    </span>
    <Flex justify="space-between" style={{ width: '100%' }}>
      <Button color="green" variant="light">
        Approved
      </Button>
      <Flex align="center" gap={16}>
        <ActionIcon color="cyan" size={24}>
          <Export size={20} />
        </ActionIcon>
        <ActionIcon color="cyan" size={24}>
          <Eye size={20} />
        </ActionIcon>
      </Flex>
    </Flex>
  </Flex>
);

export const rightMock = (
  <ActionIcon color="cyan" size={24}>
    <Export size={24} />
  </ActionIcon>
);

export const leftMock = (
  <Indicator color="red" inline offset={8} withBorder>
    <Avatar size={40}>
      <UserCircle size={40} />
    </Avatar>
  </Indicator>
);
