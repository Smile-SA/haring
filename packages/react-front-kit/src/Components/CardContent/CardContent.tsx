'use client';

import type { GroupProps, StackProps } from '@mantine/core';
import type { FlexProps } from '@mantine/core/lib/Flex/Flex';
import type { ReactElement, ReactNode } from 'react';

import { Flex, Group, Stack } from '@mantine/core';

export interface ICardContentProps extends FlexProps {
  children?: ReactNode;
  childrenGroupProps?: GroupProps;
  contentContainerFlexProps?: FlexProps;
  leftNode?: ReactNode;
  leftNodeStackProps?: StackProps;
  rightNode?: ReactNode;
  rightNodeStackProps?: StackProps;
  titleNode?: ReactNode;
  titleNodeGroupProps?: GroupProps;
  topNode?: ReactNode;
  topNodeGroupProps?: GroupProps;
}

export function CardContent(props: ICardContentProps): ReactElement {
  const {
    children,
    childrenGroupProps,
    contentContainerFlexProps,
    leftNode,
    leftNodeStackProps,
    rightNode,
    rightNodeStackProps,
    titleNode,
    titleNodeGroupProps,
    topNode,
    topNodeGroupProps,
    ...containerFlexProps
  } = props;

  return (
    <Flex
      align="stretch"
      gap="md"
      justify="stretch"
      wrap="nowrap"
      {...containerFlexProps}
    >
      {Boolean(leftNode) && (
        <Stack align="stretch" justify="flex-start" {...leftNodeStackProps}>
          {leftNode}
        </Stack>
      )}
      {Boolean(topNode || titleNode || children) && (
        <Flex
          align="stretch"
          direction="column"
          gap={4}
          justify="start"
          style={{ flexGrow: 1 }}
          {...contentContainerFlexProps}
        >
          {Boolean(topNode) && <Group {...topNodeGroupProps}>{topNode}</Group>}
          {Boolean(titleNode) && (
            <Group {...titleNodeGroupProps}>{titleNode}</Group>
          )}
          {Boolean(children) && (
            <Group style={{ flexGrow: 1 }} {...childrenGroupProps}>
              {children}
            </Group>
          )}
        </Flex>
      )}
      {Boolean(rightNode) && (
        <Stack align="stretch" justify="flex-start" {...rightNodeStackProps}>
          {rightNode}
        </Stack>
      )}
    </Flex>
  );
}
