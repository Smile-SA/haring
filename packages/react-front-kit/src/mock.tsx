/* eslint-disable react-refresh/only-export-components */

export * from './Components/ResponsiveTabs/ResponsiveTabs.mock';
export * from './Components/SidebarMenu/SidebarMenu.mock';
export * from './Components/SwitchableView/SwitchableView.mock';
export * from './Components/Thumbnail/Thumbnail.mock';
export * from './Components/ThumbnailGrid/ThumbnailGrid.mock';
export * from './Layouts/FoldableColumnLayout/FoldableColumnLayout.mock';
