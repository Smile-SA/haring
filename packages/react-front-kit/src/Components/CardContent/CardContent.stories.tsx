import type { Meta, StoryObj } from '@storybook/react';
import type { ReactElement } from 'react';

import { Container } from '@mantine/core';

import { CardContent as Cmp } from './CardContent';
import {
  childrenExampleMock,
  defaultCardContentMock,
  leftMock,
  otherTitleMock,
  otherTopMock,
  rightMock,
  titleMock,
  topMock,
} from './CardContent.mock';

const meta = {
  component: Cmp,
  tags: ['autodocs'],
  title: '3-custom/Components/CardContent',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const CardContent: IStory = {
  args: defaultCardContentMock,
};

const container = ({ ...props }): ReactElement => (
  <Container bg="white" size="400px">
    <Cmp {...props} />
  </Container>
);

export const ChildrenExample: IStory = {
  args: {
    children: childrenExampleMock,
    titleNode: titleMock,
    topNode: topMock,
    topNodeGroupProps: { mb: 12 },
  },
  render: ({ ...props }) => container(props),
};

export const RightNodeExample: IStory = {
  args: {
    pb: 24,
    rightNode: rightMock,
    titleNode: titleMock,
  },
  render: ({ ...props }) => container(props),
};

export const LeftNodeExample: IStory = {
  args: {
    leftNode: leftMock,
    titleNode: otherTitleMock,
    topNode: otherTopMock,
  },
  render: ({ ...props }) => container(props),
};
