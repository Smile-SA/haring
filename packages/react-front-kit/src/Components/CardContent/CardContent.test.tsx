import { renderWithProviders } from '@smile/react-front-kit-shared/test-utils';

import { CardContent } from './CardContent';
import { defaultCardContentMock } from './CardContent.mock';

describe('CardContent', () => {
  it('matches snapshot', () => {
    const { container } = renderWithProviders(
      <CardContent id="summary-card" {...defaultCardContentMock} />,
    );
    expect(container).toMatchSnapshot();
  });
});
