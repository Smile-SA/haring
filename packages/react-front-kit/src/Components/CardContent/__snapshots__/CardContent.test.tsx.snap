// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContent matches snapshot 1`] = `
<div>
  <div
    class="mantine-1nzdwls"
    id="summary-card"
  >
    <div
      class="mantine-Stack-root mantine-caxjnw"
    >
      left node
    </div>
    <div
      class="mantine-8tuotc"
      style="flex-grow: 1;"
    >
      <div
        class="mantine-Group-root mantine-k3ov3c"
      >
        top node
      </div>
      <div
        class="mantine-Group-root mantine-k3ov3c"
      >
        title node
      </div>
      <div
        class="mantine-Group-root mantine-k3ov3c"
        style="flex-grow: 1;"
      >
        children node
      </div>
    </div>
    <div
      class="mantine-Stack-root mantine-caxjnw"
    >
      right node
    </div>
  </div>
</div>
`;
